evmcore/dummy_block.go

```diff
@@ -138,14 +149,13 @@ type EvmHeaderJson struct {
 	TxHash      common.Hash      `json:"transactionsRoot" gencodec:"required"`
 	ReceiptHash common.Hash      `json:"receiptsRoot"     gencodec:"required"`
 	Bloom       types.Bloom      `json:"logsBloom"        gencodec:"required"`
-	Difficulty  *hexutil.Big     `json:"difficulty"       gencodec:"required"`
 	Number      *hexutil.Big     `json:"number"           gencodec:"required"`
```

Comment:
I think we need to preserve the difficulty field in the RPC output format, as, for instance, documented [here](https://www.quicknode.com/docs/kaia/eth_getHeaderByNumber).